### PR TITLE
Adding a spinner on the single sig approval

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: src/Angor/Client/Pages/Signatures.razor [hunk 1]
```diff
@@ -102,7 +102,19 @@
                                                     <td>@signature.TimeArrived.ToString("g")</td>
                                                     @if (signature.TimeApproved is null)
                                                     {
-                                                        <td><button class="btn btn-success" @onclick="() => ApproveSignatureCheckPassword(signature)">Approve</button></td>
+                                                        <td>
+                                                            <button class="btn btn-success" @onclick="() => ApproveSignatureCheckPassword(signature)" disabled="@signaturesRequestsApproving.ContainsKey(signature)">
```
Comment (itailiors) [inline src/Angor/Client/Pages/Signatures.razor:106]: this way only the the approve signature button you pick will be disabled, maybe we should disable all signature while one is approving?

File: src/Angor/Client/Pages/Signatures.razor [hunk 2]
```diff
@@ -408,103 +423,118 @@
...
-        
-        var words = await passwordComponent.GetWalletAsync();
-        
-        foreach (var signature in pendingSignatures)
+        try
         {
-            await PerformSignatureApproval(signature, words);
-            numOfSignaturesSigned++;
-            StateHasChanged();
+            if (words == null)
```
Comment (itailiors) [inline src/Angor/Client/Pages/Signatures.razor:508]: i'm noticing it now, words can't be null from 
`passwordComponent.GetWalletAsync();` right? thats why we removed the check before.
so we can remove the check here also

File: src/Angor/Client/Pages/Signatures.razor [hunk 3]
```diff
@@ -408,103 +423,118 @@
...
-            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+            Logger.LogError(e, e.Message);
+            notificationComponent.ShowErrorMessage(e.Message);
+        }
+        finally
+        {
+            signaturesRequestsApproving.Remove(signature);
         }
+
+        StateHasChanged();
```
Comment (itailiors) [inline src/Angor/Client/Pages/Signatures.razor:452]: makes sense that we want to ensure the state change only happens if an exception wasn't thrown, nice catch!

File: src/Angor/Client/Pages/Signatures.razor (hunk 2 above)
Comment (dangershony) [reply src/Angor/Client/Pages/Signatures.razor:508]: yes correct I will remove it

File: src/Angor/Client/Pages/Signatures.razor (hunk 3 above)
Comment (dangershony) [reply src/Angor/Client/Pages/Signatures.razor:452]: I believe there is a state change also inside the error  message component

**Review by itailiors — APPROVED** [review]

**Pull request merged**